Externals/curl/CMakeLists.txt

```diff
@@ -1,2 +1,59 @@
-include_directories(include/)
-add_subdirectory(lib)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/curl/include)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/curl/lib)
+
+set(CURL_LIBS ${MBEDTLS_LIBRARIES} zlibstatic)
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    set(use_core_foundation ON)
+
+    find_library(SYSTEMCONFIGURATION_FRAMEWORK "SystemConfiguration")
+    if(NOT SYSTEMCONFIGURATION_FRAMEWORK)
+        message(FATAL_ERROR "SystemConfiguration framework not found")
+    endif()
+
+    list(APPEND CURL_LIBS "-framework SystemConfiguration")
+endif()
+
+file(GLOB SRCS curl/lib/*.c curl/lib/vauth/*.c curl/lib/vquic/*.c curl/lib/vssh/*.c curl/lib/vtls/*.c)
```

Comment:
Should we consider using `CONFIGURE_DEPENDS` for this?

Reply:
The reason you don't usually glob for sources is because then CMake can't know when new files are added or files are deleted. That's not really a concern for an external we update once in a blue moon, as long as we also change any CMakeLists in the same commit that we update the external.

Reply:
Yeah, I was aware why, I didn't know if we wanted to handle changes from the glob.  This should be relatively rare, just thought I'd bring it up as I wasn't sure if it had been discussed before.